template/tpl.go

```diff
@@ -2,18 +2,72 @@ package template
 
 import (
 	"bytes"
+	"os"
+	"path/filepath"
 	"text/template"
 
 	"github.com/Masterminds/sprig/v3"
 )
 
-func RenderTpl(input []byte, bindings map[string]interface{}) ([]byte, error) {
-	tpl, err := template.New("gotpl").Funcs(sprig.TxtFuncMap()).Parse(string(input))
+// include function to include other templates
+func include(name string, data interface{}, tpl *template.Template, baseDir string) (string, error) {
+	var buf bytes.Buffer
+
+	// Construct the file path for the included template
+	includePath := filepath.Join(baseDir, name)
+
+	// Read and parse the included template
+	includeContent, err := os.ReadFile(includePath)
```

Comment:
is this a mirror of the implementation of include in helm?  I feel like it might work differently

Reply:
My general guess is it should really just use https://cs.opensource.google/go/go/+/refs/tags/go1.22.3:src/text/template/template.go;l=188 and then call execute on that (the expectation would be the template is defined in the file itself and so would have been parsed by the text/template library and lookup-able)

Reply:
It actually is, the confusion is because I believe helm needs to make a redefinition of that function to handle the fact they can be defined anywhere in the chart, not in-file.  If you look at the implementation and how it's used, it references a named `define` block

Reply:
Okay, after further discussion with and these PR comments from @michaeljguarino, this Polly PR needs an update to not include the referenced template file but a template mentioned from a `define` directive within the current .tpl file being rendered.
Changes are coming henceforth.